File: shoring/compat/src/main/java/com/panopset/compat/AniGrid.java
package com.panopset.compat;

public class AniGrid {

  public final int mapw;
  public final int maph;
  private final Integer h;
  private final Integer ht;
  private final Integer w;
  private Double multiplier = 1.0;

  public AniGrid(final int mapWidth, final int mapHeight, final int gridWidth, final int gridHeight,
      final double multiplier) {
    this(mapWidth, mapHeight, gridWidth, gridHeight);
    this.multiplier = multiplier;
  }

  public AniGrid(final int mapWidth, final int mapHeight, final int gridWidth,
      final int gridHeight) {
    this.mapw = mapWidth;
    this.maph = mapHeight;
    this.w = mapw / gridWidth;
    this.h = maph / gridHeight;
    Integer wt = this.w + ((int) ((this.w * .5) * 0));
    this.ht = (wt * this.h) / this.w;
  }

  public int getMultiplier() {
    return multiplier.intValue();
  }

  public int getDspWidth() {
    double rtn = multiplier * w;
    return (int) rtn;
  }

  public int getDspHeight() {
    double rtn = multiplier * ht;
    return (int) rtn;
  }

  public int[] getPaintDimensions(final int gridx, final int gridy, final int x, final int y) {
    int sx1 = gridx * w;
    int sy1 = gridy * h;
    return new int[] {sx1, sy1, w, h, x, y, getDspWidth(), getDspHeight()};
  }

}

File: code/shoring/compat/src/main/kotlin/com/panopset/compat/AniGrid.kt
package com.panopset.compat

class AniGrid(
    mapw: Int, maph: Int, gridWidth: Int,
    gridHeight: Int
) {
    private val h: Int
    private val ht: Int
    private val w: Int
    var multiplier = 1.0

    constructor(
        mapWidth: Int, mapHeight: Int, gridWidth: Int, gridHeight: Int,
        multiplier: Double
    ) : this(mapWidth, mapHeight, gridWidth, gridHeight) {
        this.multiplier = multiplier
    }

    init {
        w = mapw / gridWidth
        h = maph / gridHeight
        val wt = w + (w * .5 * 0).toInt()
        ht = wt * h / w
    }

    val dspWidth: Int
        get() {
            val rtn = multiplier * w
            return rtn.toInt()
        }
    val dspHeight: Int
        get() {
            val rtn = multiplier * ht
            return rtn.toInt()
        }

    fun getPaintDimensions(gridx: Int, gridy: Int, x: Int, y: Int): IntArray {
        val sx1 = gridx * w
        val sy1 = gridy * h
        return intArrayOf(sx1, sy1, w, h, x, y, dspWidth, dspHeight)
    }
}
